hmt_escrow/storage.py

```diff
@@ -56,19 +88,35 @@ def download(key: str, private_key: bytes) -> Dict:
         Exception: if reading from IPFS fails.
 
     """
-    try:
-        LOG.debug("Downloading key: {}".format(key))
-        ciphertext = IPFS_CLIENT.cat(key, timeout=30)
-        msg = _decrypt(private_key, ciphertext)
-    except Exception as e:
-        LOG.warning(
-            "Reading the key {!r} with private key {!r} with IPFS failed because of: {!r}"
-            .format(key, private_key, e))
-        raise e
-    return json.loads(msg)
-
-
-def upload(msg: Dict, public_key: bytes) -> Tuple[str, str]:
+    if key.startswith("s3"):
+        try:
+            LOG.debug("Downloading s3 key: {}".format(key))
+            BOTO3_CLIENT = _connect_s3()
+            response = BOTO3_CLIENT.get_object(Bucket=ESCROW_BUCKETNAME,
```

Comment:
might be handy to break these into two separate try/catches, one that does the fetching, one that does the decrypting, so the log message is targeted.  not blocking though